MID Server/MID Server Script Includes/IllumioHTTPClient.js

```diff
@@ -98,60 +98,47 @@ IllumioHTTPClient.prototype = {
             }
 
             // if response status is 200 ok or 202 for async request
-            if (SUCCESS_CODES.indexOf(responseCode) != -1) {
+            if (responseCode >= 200 && responseCode < 300) {
                 var responseObj = {
                     hasError: false,
                     data: new JSON().decode(responseBody),
                     headers: responseHeaders,
                     status: responseCode
                 };
-
+                this.logger._debug('HTTP ' + method.getName() + ' Call complete, status: ' + responseCode);
                 return responseObj;
             }
-            if (retryCount == 0 || (RETRY_CODES.indexOf(responseCode) == -1 && !(responseCode >= 500))) {
+
+            retryCount = (typeof retryCount == 'undefined') ? this.httpRetryCount : retryCount;
+            if (retryCount <= 0 || (RETRY_CODES.indexOf(responseCode) == -1 && !(responseCode >= 500))) {
+                this.logger._error('Failed: HTTP' + method.getName() + ' Call, status: ' + responseCode);
                 return {
                     hasError: true,
                     status: responseCode
                 };
             }
 
             lastSleepTime = lastSleepTime ? lastSleepTime : 0;
-            var currentSleepTime = Math.min(this.HTTP_RETRY_INTERVAL_MAX, lastSleepTime + this.HTTP_RETRY_INTERVAL_INCREMENT);
-            lastSleepTime = currentSleepTime;
-            this.logger._error('Failed to execute HTTP call: Waiting for ' + currentSleepTime + ' seconds before doing another REST call.');
-            this._sleep(currentSleepTime * 1000);
+            lastSleepTime = Math.min(this.httpRetryIntervalMax, lastSleepTime + this.httpRetryIntervalIncrement);
+            if (responseCode == 429) {
+                if (responseHeaders['retry-after']) {
+                    lastSleepTime = parseInt(responseHeaders['retry-after']);
+                }
+            }
+            this.logger._error('Failed to execute HTTP call: Waiting for ' + lastSleepTime + ' seconds before doing another REST call.');
+            Packages.java.lang.Thread.sleep(lastSleepTime * 1000);
```

Comment:
If this is working, can we remove the commented out line below and the `_sleep` function?